hydro_lang/src/keyed_stream.rs

```diff
@@ -644,7 +685,10 @@ where
         self,
         init: impl IntoQuotedMut<'a, I, L>,
         comb: impl IntoQuotedMut<'a, F, L>,
-    ) -> KeyedSingleton<K, A, L, B> {
+    ) -> KeyedSingleton<K, A, L, B>
+    where
+        B: Boundedness,
```

Comment:
Why do we need this where clause (here and below)? I think it is already in the impl constraints?